tests/test_Jetscape.py

```diff
@@ -499,3 +499,24 @@ def test_Jetscape_read_parton_file(jetscape_file_path_partons):
     # Test construction with wrong particletype (not string)
     with pytest.raises(TypeError):
         Jetscape(jetscape_file_path_partons, particletype=1)
+
+def test_update_after_merge_warning(jetscape_file_path):
+    # Create two Jetscape instances with different particle types and defining strings
+    jetscape1 = Jetscape(jetscape_file_path)
```

Comment:
Instead of setting the particle type variables by hand, you could do it in the constructor and put "parton" and "hadron" there. These are the two supported types.

Reply:
But Jetscape has no field for this in the constructor?